models/event.js

```diff
@@ -1 +1,96 @@
+	address: { type: String, required, minLength: 3, maxLength: 100 },
+	category: { 
+        // type: [ { type: String, enum: ["Free", "Music", "Workshop", "Film", "Family", "Market", "Exhibition", "Party", "Performing Arts", "Opening", "Food", "Reading"] } ], required // found this here, apparently then multi-select works for enums: https://stackoverflow.com/questions/27447876/is-it-possible-to-create-a-multi-select-enum-in-mongoose
+        type: String,
+        enum: ["Free", "Music", "Workshop", "Film", "Family", "Market", "Exhibition", "Party", "Performing Arts", "Opening", "Food", "Reading"],
+        required 
+    },
+    summary: { type: String, required, minLenth: 3, maxLength: 100 },
+    description: { type: String, required, minLength: 3, maxLength: 1000 },
+
+    // Date and Time 
+    // min 3 day ahead, max 6 months to the future
+    start_date: { 
+        type: Date, 
+        min: date.add(3, "days"), 
+        max: date.add(6, "months")  
+    }, // date.add(3, "days").format("dddd, DD MMMM YYYY") prints "Friday, 26 November 2021" // see test.js for example how to use moment.js
+    end_date: { 
+        type: Date, 
+        min: date.add(3, "days"), 
+        max: date.add(6, "months") 
+    },
+    start_time: {
+        type: Date, 
+        min: date.add(3, "days"), 
+        max: date.add(6, "months")
+        // date.add(3, "days").format("dddd, DD MMMM YYYY") prints "13:21" 
+    },
+    end_time: {
+        type: Date, 
+        min: date.add(3, "days"), 
+        max: date.add(6, "months") 
+    }, 
+    happens_once: { type: Boolean },
+    happens_daily_or_weekly: { type: String, enum: ["daily", "weekly"] },
+    happens_on_specific_dates: [ { 
+        type: Date, 
+        min: date.add(3, "days"), 
+        max: date.add(6, "months") 
+    } ], // can you do multiple dates like this?
+
+    // Prices and Booking
+    free_event: { type: Boolean, required },
+    price: { type: Number },
+    booking_required: { type: Boolean, required },
+    booking_site: { type: String },
+
+    // Contact Details
+    website: { type: String, minLength: 5 },
+    email: { type: String },
+    phone: { type: String, minLength: 6, maxLength: 15 }, // ?? more validation
+    instagram: { type: String, minLength: 3 },
+    twitter: { type: String }, // this might be deleted
+    facebook: { type: String, minLength: 3 },
+
+    // Image
+    image: { type: Schema.Types.ObjectId, ref: "images"
+        // event.path(image).get(value => `${rootImageUrl}/${value}`); 
+    },
+
+    // Review
+    draft: { type: Boolean, default: true },
+    submitted: { type: Boolean, default: false },
+    reviewed: { type: Boolean, default: false },
+    published: { type: Boolean, default: false },
+
+    // for CRUD actions on event listings
+    author: { type: Schema.Types.ObjectId, ref: "regUsers" }, // user that created the event
+    created_at: { type: Date, default: date },
```

Comment:
The "default" should either be a static value like the string "potato" or a reference to a function like "getDate" or "Date.now" -- if it's "potato" then all the fields will be "potato" by default. If it's "Date.now" then the function "Date.now()" will be run every time when a new instance of this model is created.

Currently when you make a variable "date" and use that, it's kind of like the "potato" example and new instances of Event will have the same value. The "date" variable gets its value when Node is first interpreting this file and that value will be kept until Node restarts, so it's not actually quite perfect here :)

Reply:
oh, so i better use moment.utc() in all cases, i want the current time in there, correct?